lib/jsdom/browser/history.js

```diff
@@ -11,10 +11,10 @@ function StateEntry(data, title, url) {
 module.exports = History;
 
 function History(window) {
-  this._states = [new StateEntry(null, "", window.location._url.href)];
+  this._states = [new StateEntry(null, "", window._document._URL)];
   this._index = 0;
   this._window = window;
-  this._location = window.location;
+  this._location = window._document._location;
```

Comment:
Because that's where it's really stored; window.location is a getter for this._document._location